addons/purchase_mrp/models/purchase.py

```diff
@@ -18,12 +18,18 @@ class PurchaseOrder(models.Model):
     @api.depends('order_line.move_dest_ids.group_id.mrp_production_ids')
     def _compute_mrp_production_count(self):
         for purchase in self:
-            purchase.mrp_production_count = len(purchase.order_line.move_dest_ids.group_id.mrp_production_ids |
-                                                purchase.order_line.move_ids.move_dest_ids.group_id.mrp_production_ids)
+            purchase.mrp_production_count = len(purchase._get_mrp_productions())
+
+    def _get_mrp_productions(self, remove_archived_picking_types=True):
```

Comment:
It would be cleaner to use a kwargs there.
```python3
def _get_mrp_productions(self, **kwargs):
    return self.order_line.move_dest_ids.group_id.mrp_production_ids | self.order_line.move_ids.move_dest_ids.group_id.mrp_production_ids
```
in mrp_subcontracting_purchase
```python3
def _get_mrp_productions(self, **kwargs):
    production = super()._get_mrp_productions(kwargs)
    if kwargs.get('remove_archived_picking_types', True):
        productions = productions.filtered(lambda production: production.with_context(active_test=False).picking_type_id.active)
    return productions
```

Reply:
done
mrp_subcontracting_purchase now depends on purchase_mrp...